pkg/api/dirs.go

```diff
@@ -94,6 +95,8 @@ func (s *Service) dirUploadHandler(w http.ResponseWriter, r *http.Request, store
 			jsonhttp.PaymentRequired(w, "batch is overissued")
 		case errors.Is(err, errEmptyDir):
 			jsonhttp.BadRequest(w, errEmptyDir)
+		case err.Error() == errInvalidTar.Error():
```

Comment:
can this be replaced by `errors.Is(err, errInvalidTar)` ?

Reply:
This doesn't work somehow. I tried it.

Reply:
The `errors.Is` doesnt work as this errInvalidTar is not actually returned from the function. It is a hard-coded string error. This is not the right way to do error checking.

https://pkg.go.dev/archive/tar#pkg-variables

You need to check `errors.Is(err, tar. ErrHeader)`

Reply:
If you were looking at errEmptyDir, it is actually being returned below(L206). This is why the check works.

Reply:
@aloknerurkar Thanks. Now it's better.